src/Asset/EntrypointsJsonLookup.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+/*
+ * Copyright (c) 2018 Heimrich & Hannot GmbH
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace HeimrichHannot\EncoreBundle\Asset;
+
+
+use Symfony\Component\DependencyInjection\Container;
+
+class EntrypointsJsonLookup
+{
+
+    public function mergeEntries(array $entrypointsJsons, array $entries, string $babelPolyfillEntryName = null) : array
+    {
+        foreach ($entrypointsJsons as $entrypointsJson) {
+            $entrypoints = $this->parseEntrypoints($entrypointsJson);
+
+            $entriesMap = [];
+            foreach ($entries as $entry) {
+                if (!isset($entry['name'])) {
+                    continue;
+                }
+
+                $entriesMap[$entry['name']] = true;
+            }
+
+            foreach ($entrypoints as $name=>$entrypoint) {
+                // Ignore the babel-polyfill entry
+                if ($name == $babelPolyfillEntryName) continue;
+
+                // Only add entries that not already exist in the symfony config
+                if (!isset($entriesMap[$name])) {
+                    $newEntry = [
+                        'name' => $name,
+                        'head' => false,
+                    ];
+
+                    if (isset($entrypoint['css'])) {
+                        $newEntry['requiresCss'] = true;
+                    }
+
+                    $entries[] = $newEntry;
+                }
+            }
+        }
+
+        return $entries;
+    }
+
+    public function parseEntrypoints(string $entrypointsJson) : array
+    {
+        if (!file_exists($entrypointsJson)) {
```

Comment:
I think we should do some caching here, as the entrypoints.json file will be read and parsed on every page load. Don't know if this is the best point for this, but it should be at least before the file reading and parsing. Could be done in production mode only.

Reply:
I actually already thought about that and then forgot it. I just looked at the Webpack Encore Bundle again. They already parsing this file and do some caching for it. The "right" way would be to file a ticket against that bundle to expose the entries via the EntrypointLookup service.

Maybe we can add something temporary.

Reply:
Maybe you could use the entryChoice class or create an own Choice class for this. They have already an enhance caching mechanism.

Reply:
I added caching via the caching of the Webpack Encore Bundle. It is not very pretty but should work. Please remember that caching in the Webpack Encore Bundle is not activated by default.

Reply:
Looks good for me.